tests/garage/tf/regressors/test_gaussian_mlp_regressor_with_model.py

```diff
@@ -0,0 +1,103 @@
+from nose2.tools.params import params
+import numpy as np
+import tensorflow as tf
+
+from garage.tf.optimizers import PenaltyLbfgsOptimizer
+from garage.tf.regressors import GaussianMLPRegressorWithModel
+from tests.fixtures import TfGraphTestCase
+
+
+class TestGaussianMLPRegressorWithModel(TfGraphTestCase):
+    def test_fit(self):
+        gmr = GaussianMLPRegressorWithModel(input_shape=(1, ), output_dim=1)
+        data = np.linspace(-np.pi, np.pi, 1000)
+        obs = [{'observations': [[x]], 'returns': [np.sin(x)]} for x in data]
+
+        observations = np.concatenate([p['observations'] for p in obs])
+        returns = np.concatenate([p['returns'] for p in obs])
+        for _ in range(100):
+            gmr.fit(observations, returns.reshape((-1, 1)))
+
+        paths = {
+            'observations': [[-np.pi], [-np.pi / 2], [-np.pi / 4], [0],
+                             [np.pi / 4], [np.pi / 2], [np.pi]]
+        }
+
+        prediction = gmr.predict(paths['observations'])
+
+        expected = [[0], [-1], [-0.707], [0], [0.707], [1], [0]]
+        assert np.allclose(prediction, expected, rtol=0, atol=0.1)
+
+    def test_fit_custom(self):
+        gmr = GaussianMLPRegressorWithModel(
+            input_shape=(1, ),
+            output_dim=1,
+            subsample_factor=0.9,
+            normalize_inputs=False,
+            normalize_outputs=False)
```

Comment:
Emm is this test legit? I don’t see [this line](https://github.com/rlworkgroup/garage/pull/606/files#diff-1d908bac267c9cae4bb670633344a07eR193) is checked here. I am not clear the purpose of this test.

Reply:
`normalize_inputs` and `normalize_outputs` are by default True. This test is for testing when they are False.

Reply:
Yes I am aware of it. I meant that I am confused because in this test you don’t check that line or check if the outputs is correctly unnormalized.

Reply:
I see. The purpose of this test is to see if it can still fit properly. You are right we should assert as much as we can.

Reply:
I guess a better question is I am not sure what this test is trying to check. A state check or a interaction check? Is the purpose to check the fit result is correct with customized parameters? Then it can be many combinations, why not split it to one unnormalized test and one `subsample_factor=.9`?

Reply:
Yes it makes sense. Separating into smaller checks, and also assert about something. I will modify that.